public/app/features/correlations/Forms/QueryEditorField.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { Controller } from 'react-hook-form';
+import { useAsync } from 'react-use';
+
+import { getDataSourceSrv } from '@grafana/runtime';
+import { Field, LoadingPlaceholder, Alert } from '@grafana/ui';
+
+interface Props {
+  dsUid?: string;
+  name: string;
+  invalid?: boolean;
+  error?: string;
+}
+
+export const QueryEditorField = ({ dsUid, invalid, error, name }: Props) => {
+  const {
+    value: datasource,
+    loading: dsLoading,
+    error: dsError,
+  } = useAsync(async () => {
+    if (!dsUid) {
+      return;
+    }
+    return getDataSourceSrv().get(dsUid);
+  }, [dsUid]);
+  const QuerEditor = datasource?.components?.QueryEditor;
+
+  return (
+    <Field label="Query" invalid={invalid} error={error}>
+      <Controller
+        name={name}
+        // FIXME: this won't be needed when the API will return the default config
```

Comment:
I would prefer to cross-reference it to make sure it's not forgotten. I've created a clean-up issue: https://github.com/grafana/grafana/issues/55737
```suggestion
        // TODO: #55737 this won't be needed when the API will return the default config
```